### ISSUE TITLE
Extreme Bevy 3.5: Detecting Desyncs

### ISSUE THREAD

**Issue by johanhelsing** [johanhelsing/johanhelsing.github.io#37]
https://johanhelsing.studio/posts/extreme-bevy-desync-detection

**Comment by axelmagn** [comment]
I ran into a curious issue in the middle of this section when I implemented `start_synctest_session` (end of "Back to synctest sessions"). When starting in synctest mode, the players are created and then immediately deleted.  It happens quickly enough that my logging never registers an existing player during the `move_players` step.

I suspect that it's an issue with ggrs rollback and how it interacts with synctest, but I'm not familiar enough with the library to say.   I also have noticed throughout the tutorial sometimes code from "later" sections sometimes sneak into earlier code snippets (example: definition of `GameState` in Part 3).  So maybe there is some code that I was not meant to include yet.

At any rate, I wanted to give you a heads up, because otherwise I really like this tutorial and I'm glad it exists.  I'll probably try to just charge ahead through part 3.5 without trying to pass all the checkpoints.  I generally don't upload my tutorial code, but lmk if it would be helpful for repro on your side and I'll set something up.  Keep up the good work.

**Comment by langep** [comment]
I was not able to reproduce the desync bug mentioned in the article at first and had to do some digging. Synctest will simulate a rollback every frame (by default the last two states) and we use rollback registered entities and our own transform checksum to determine a desync. Now, if one player shoots another but the despawned player does not move, there we won't detect a desync because the desync happens when when a despawned player should have moved instead (i.e. get a different transform). 

TL;DR: The killed player has to be moving on the frame (or one frame before) it is hit by the bullet otherwise no desync detected. If we extend the simulated rollback distance on the session, any movement in the rollback simulated windows should also be enough to cause a desync.
